File: app/models/ring-assignment.js
import DS from 'ember-data';
const { Model, attr, belongsTo } = DS;

export default class RingAssignmentModel extends Model {
  @belongsTo('competition') competition;
  @belongsTo('robot') robot;
  @attr('number') ring;
  @attr('string') letter;
  @attr('number') round;
};

File: pod/flask/python/RingAssignment.py
class RingAssignment(object):
    def __init__(
        self,
        id: int = 0,
        competition: str = "",
        robot: int = -1,
        letter: str = "",
        placed: int = -1,
        ring: int = -1,
    ):
        self.id = id
        self.competition = competition
        self.robot = robot
        self.letter = letter
        self.placed = placed
        self.ring = ring

    def __eq__(self, other):
        return self.id == other.id
